### PR TITLE
board/cpu init: move cpu_init() from board_init() to startup

### PR DESCRIPTION
**Pull request by smlng**
### Contribution description

During "boot" of RIOT the initial startup (reset) function defined per CPU calls `board_init()` which then calls `cpu_init()` and maybe inits some LEDs and buttons afterwards. IMHO its a bit weird that `board_init` calls `cpu_init` instead of handling this in the startup function by calling `board_init` after `cpu_init`.

This PR moves the `cpu_init` function accordingly. I tested with Arduinos, PhyNode and Remote-RevB, which all work. However, some boards have very special startup routines so if anybody could check the following boards to verify they still work, would be great:

- [x] mbed_lpc1768
- [ ] msba2
- [ ] mulle (@gebart ?)
- [x] seeeduino_arch-pro
- [ ] waspmote-pro
- [ ] msp430 (on hold, not changed yet)
  - [ ] chronos
  - [ ] msb-430
  - [ ] telosb

### CODE REVIEW

**Comment by kaspar030** [comment]
> +24 −661

Awesome! Something like this was long overdue.

**Comment by basilfx** [comment]
I can do the Seeeduino Arch-Pro and the mbed LPC1768.

**Comment by aabadie** [comment]
Nice one!

**Comment by smlng** [comment]
The msp430 based are bit complicated to sort out, that's why I reverted the changes there for now.

File: boards/mbed_lpc1768/system.c [hunk 1]
```diff
@@ -1,478 +0,0 @@
-/**************************************************************************//**
```
Comment (jia200x) [inline boards/mbed_lpc1768/system.c]: why is this file removed?

Comment (smlng) [reply boards/mbed_lpc1768/system.c]: oops, just for fun? 😄

Comment (jia200x) [reply boards/mbed_lpc1768/system.c]: sounds good :)

Comment (smlng) [reply boards/mbed_lpc1768/system.c]: ah no, because its basically the same file as for the seeedoino and that I moved to the cpu/lpc1768 so its shared with the mbed board, hence the remove.

Comment (jia200x) [reply boards/mbed_lpc1768/system.c]: ok, cool! :)

**Comment by basilfx** [comment]
mbed LPC1768 and Seeeduino Arch-Pro seem to work (tested `examples/default`).

**Comment by tcschmidt** [comment]
@jia200x any changes to request? Or is the review completed?

**Comment by smlng** [comment]
please note, this is still WIP (see label) need to sort out the msp430 based boards, i.e., don't merge
